.github/workflows/build-deploy-middleware.yml

```diff
@@ -0,0 +1,76 @@
+name: Create, publish, deploy a Middleware API image
```

Comment:
Do we need this separate workflow?  It looks like we are doing this in the `deploy-dev` workflow.  Perhaps we could simply have it in `deploy-dev` unless it will inherit the results from this new workflow.  I am thinking about how to make things clean and seamless for the handoff.

Reply:
The separate workflows are so we can build and deploy each service separately so we dont have to always rebuild everything at once in return enabling faster deployments and changes to the environments.